metriq_gym/qplatform/device.py

```diff
@@ -63,3 +63,18 @@ def _(device: LocalAerDevice) -> rx.PyGraph:
     if coupling_list is None:
         return rx.generators.complete_graph(device._backend.configuration().n_qubits)
     return coupling_map_to_graph(CouplingMap(coupling_list))
+
+
+@connectivity_graph.register
+def _(device: QuantinuumDevice) -> rx.PyGraph:
+    df = qnx.devices.get_all(issuers=[qnx.devices.IssuerEnum.QUANTINUUM]).df()
+    df["n_qubits"] = [
+        len(bi.architecture.nodes) if hasattr(bi.architecture, "nodes")
+        else (_ for _ in ()).throw(RuntimeError("Could not determine qubit count"))
+        for bi in df["backend_info"]
+    ]
+    device_name = device.profile.device_id
+    row = df[df["device_name"] == device_name].iloc[0]
+    num_qubits = row["n_qubits"]
```

Comment:
I think it makes more sense to slice horizontally by device_name first, and then fetch the number of qubits.
Also, I would use the information from the architecture field to check whether the device is fully connected before generating a complete graph.